contracts/transfer/circuits/src/gadgets/commitment.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details. 
+
+// use phoenix_core::{Note, TransactionItem, ViewKey};
+
+use dusk_plonk::constraint_system::ecc::scalar_mul::fixed_base::scalar_mul;
+use dusk_plonk::jubjub::{
+    Fr, AffinePoint, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED, ExtendedPoint,
+};
+use dusk_plonk::prelude::*;
+use rand::*;
+use plonk_gadgets::AllocatedScalar;
+use dusk_bls12_381::Scalar;
+
+// pub struct PedersenCommitment {
+//     r: ExtendedPoint
+//     c: ExtendedPoint
+// }
+
+/// Prove knowledge of the value and blinding factor, which make up the value commitment.
+/// This commitment gadget is using the pedersen commitments.
+pub fn commitment(composer: &mut StandardComposer, value: Fr, blinder: Fr, pub_commit: AffinePoint) {
```

Comment:
Again, you should recieve variables not scalars. Otherways you're not mantaining the links over the gadgets and the CS.